app/models/request.rb

```diff
@@ -411,27 +411,11 @@ def aliquot_attributes
     { study_id: initial_study_id, project_id: initial_project_id, request_id: id }
   end
 
-  def get_value(request_information_type)
-    return '' unless request_metadata.respond_to?(request_information_type.key)
-
-    value = request_metadata.send(request_information_type.key)
-    return value.to_s if value.blank? || (request_information_type.data_type != 'Date')
-
-    value.to_date.strftime('%d %B %Y')
-  end
-
-  def value_for(name, batch = nil)
-    rit = RequestInformationType.find_by(name: name)
-    rit_value = get_value(rit) if rit.present?
-    return rit_value if rit_value.present?
-
-    event_value_for(name, batch)
-  end
-
-  def event_value_for(name, batch = nil)
-    list = (batch.present? ? lab_events_for_batch(batch) : lab_events)
-    list.each { |event| desc = event.descriptor_value_for(name) and return desc }
-    ''
+  def detect_descriptor(name, descriptor_batch: batch)
```

Comment:
Moved in from the FlowcellIO. Couple of adjustments:
1) Filters to the current batch by default. Only likely to have an impact of a request is moved between batches, but that does happen, and there are stories in the backlog to make it easier.
2) Uses the lab_events_for_batch to to the initial sorting to avoid repeating ourselves.